src/com/amazon/ion/impl/_Private_IonTextWriterBuilder.java

```diff
@@ -73,6 +74,7 @@ private _Private_IonTextWriterBuilder(_Private_IonTextWriterBuilder that)
         this._blob_as_string      = that._blob_as_string     ;
         this._clob_as_string      = that._clob_as_string     ;
         this._decimal_as_float    = that._decimal_as_float   ;
+        this._float_nan_and_inf_as_null = that._float_nan_and_inf_as_null;
```

Comment:
Can a user ever set this to `false` and if not, what is the point of capturing an option?  What does `true` mean here?  Does it mean invalid JSON or generate a `string` value?

Reply:
A user cannot set it _directly_. I know it seems a little strange to model this as an option that the user cannot directly set, but there is no generic "json" option. Any existing option that could be used as an indicator of writing json is already named for a specific (and different) ion type, and I think it would be bad/confusing to make the `writeFloat` method dependent on an option that looks like it's being used in the wrong spot.

Setting this value to true means that `nan` and `+/-inf` will be written as `null`. If it were to be set to false when writing json, it would result in invalid json (as is happening now).

Reply:
I guess my question is, why do we have an option that allows for invalid JSON?  And if there is an option, how would a user set it to get the broken JSON mode?

Reply:
Okay. Now I understand your concern. This class (`_Private_IonTextWriterBuilder`) is used for building json _and_ regular Ion writers, which is why the option needs to be there. This particular option is set to `false` for a standard Ion text writer. It gets set to `true` _if and only if_ the user is trying to write json. Ie.
```java
IonTextWriterBuilder.json() //...
IonTextWriterBuilder.standard().withJsonDowngrade() //...
```

Technically, it is possible for a user to override this option to create invalid json, but it's not something you would really do by accident because it involves casting to one of the "private"/internal `ion-java` classes. Ie.
```java
((_Private_IonTextWriterBuilder) options)._float_nan_and_inf_as_null = false;
```

Reply:
Ah okay--this makes sense--since it is used by both code paths, we really should add a comment there that makes it more obvious, thanks for bearing with me.